pkg/rid/cockroach/subscriptions.go

```diff
@@ -159,45 +178,28 @@ func (c *SubscriptionStore) InsertSubscription(ctx context.Context, s *ridmodels
 		cids[i] = int64(cell)
 	}
 
-	var err error
-	var ret *ridmodels.Subscription
-	if s.Version.Empty() {
-		ret, err = c.processOne(ctx, insertQuery,
-			s.ID,
-			s.Owner,
-			s.URL,
-			s.NotificationIndex,
-			pq.Int64Array(cids),
-			s.StartTime,
-			s.EndTime)
-	} else {
-		ret, err = c.processOne(ctx, udpateQuery,
-			s.ID,
-			s.Owner,
-			s.URL,
-			s.NotificationIndex,
-			pq.Int64Array(cids),
-			s.StartTime,
-			s.EndTime,
-			s.Version.ToTimestamp())
-	}
-	return ret, err
+	return c.processOne(ctx, insertQuery,
+		s.ID,
+		s.Owner,
+		s.URL,
+		s.NotificationIndex,
+		pq.Int64Array(cids),
+		s.StartTime,
+		s.EndTime)
 }
 
-// DeleteSubscription deletes the subscription identified by "id" and
-// returns the deleted subscription.
-func (c *SubscriptionStore) DeleteSubscription(ctx context.Context, s *ridmodels.Subscription) (*ridmodels.Subscription, error) {
+// UnsafeDeleteSubscription deletes the subscription identified by ID.
+// It must be done in a txn and the version verified.
+func (c *SubscriptionStore) UnsafeDeleteSubscription(ctx context.Context, s *ridmodels.Subscription) (*ridmodels.Subscription, error) {
```

Comment:
Shouldn't the signature be reduced to the ID rather than the whole Subscription?  (to match with comments above, which it seems accurately capture the intent)